src/java.desktop/share/classes/java/awt/font/NumericShaper.java

```diff
@@ -1522,7 +1531,7 @@ public static NumericShaper getContextualShaper(Set<Range> ranges,
      */
     private NumericShaper(int key, int mask) {
         this.key = key;
-        this.mask = mask;
+        this.mask = mask & (CONTEXTUAL_MASK | ALL_RANGES);
```

Comment:
Note that this will accept the CONTEXTUAL_MASK even if it was provided by the app. It is unclear whether we should discard it in that case or not.